src/mgr/OSDPerfMetricReport.h

```diff
@@ -0,0 +1,27 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#ifndef OSD_PERF_METRIC_REPORT_H_
+#define OSD_PERF_METRIC_REPORT_H_
+#include "include/denc.h"
+
+#include "common/perf_counters.h"
+
+struct PerformanceCounterDescriptor {
+  std::string name;
+  perfcounter_type_d type;
+};
+
+
+struct OSDPerfMetricReport {
+  std::vector<PerformanceCounterDescriptor> performance_counter_descriptors;
```

Comment:
@dillaman I am trying to apply this on osd side and have questions.

Do we really need to provide the descriptors in the report? The list is based on the query, the mgr should know which report to which query belongs so should be able to build this list itself.

And another question: how are we going to work with this performance counter descriptors list? There should be some rules so the osd could map a descriptor to statistics it collects. Do we want it to be just 1 to 1 mapping of predefined counters (like `inb`, `outb`, `latency`) or do we want to allow to specify more complicated rules (like `sqrt(inb + outb)`)? The second looks a little complicated. For the firts, it looks like we don't need `std::string name` in `PerformanceCounterDescriptor` but `enum` instead to specify what exactly osd counter we mean.

And the simplest (though not optimal) solution would be just to return all available counters for all queries (`op`, `inb`, `outb`, `latency` probably would be enough).

Reply:
@trociny It's mostly about forward and backwards compatibility. Yes, you could have an assumption that the OSD and MGR will always agree on what metrics should be included and their order, but things might change in the future. Indeed, an enum could be used in place of the string since that should allow more than enough room for change. I think we can stick w/ the fixed set of (```write ops```, ```read ops```, ```write bytes```, ```read bytes```, ```write latency```, and ```read latency```) for starters (I think it's important to separate read vs write latencies since they have very different expectations).

Reply:
I'll change the `std::string name` for an `enum` then, with the above set of metrics